### ISSUE TITLE
Spam Skill on NP Turn

### ISSUE THREAD

**Issue by Patcheresu** [Fate-Grand-Automata/FGA#903]
### Preparation

- [X] No one has asked for this feature before in other issues (even the closed ones)

### Describe the feature

The option to have certain skills only be inputted when the NP gauge of the relevant servant is 100%.

Say I have a servant who relies on 1 turn buffs to be used with their NPs. Under Spam rules they'll use it turn 1 no matter what or whenever the software demands they use it. Under Danger they'll use it Defensively. But there isn't a good Offensive option for 1 turn buff skills like say Chloe's Skill 2 or Monstrous Strength on servants like Astolfo and Kintoki Berserker.

**Comment by MathewSachin** [comment]
This would be cool to have, but, the main blocker is that we don't have a way detect if NP is available right now.
We can't match text since the app doesn't have text recognition, only have image recognition available.

The only weird way I can think of is to match the right edge of NP bar with an image of not-fully-charged bar, since we also need to detect overcharge. Will have to experiment with 98% charged NP for that (not sure if 99% is possible) which would be a pain to setup.

**Comment by MathewSachin** [comment]
If we're ever able to detect if NP is not fully charged, we can also add a 'Spam to charge NP' option.

**Comment by Patcheresu** [comment]
Would it be impossible to identify the condition using the glowing yellow border around the NP meter or a lack of it?

**Comment by MathewSachin** [comment]
Nah, the matching algo isn't smart enough to distinguish glows unless it is a really big static region.

**Comment by reconman** [comment]
99 % is only possible right now by charging to over 100 % and then triggering a skill which reduces the NP bar again. At least afaik.

For example, charge Parvati to 109 % and then use her NP charge skill on someone else.
